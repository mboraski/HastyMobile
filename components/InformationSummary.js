// 3rd Party Libraries
import React, { Component } from 'react';
import { StyleSheet, View, Text, ScrollView } from 'react-native';
import { Button } from 'react-native-elements';
import { Field } from 'redux-form';

// Relative Imports
import InlineLabelTextInputField from './InlineLabelTextInputField';
import Radio from './Radio';
import Color from '../constants/Color';
import { emY } from '../utils/em';

class InformationSummary extends Component {
    render() {
        return (
            <ScrollView style={styles.container}>
                <Text style={styles.formSection}>INFORMATION</Text>
                <View style={styles.formInputs}>
                    <InlineLabelTextInputField
                        name="name"
                        label="Full Name"
                    />
                    <InlineLabelTextInputField
                        name="number"
                        label="Phone Number"
                    />
                    <InlineLabelTextInputField
                        name="email"
                        label="Email"
                        keyboardType="email-address"
                    />
                </View>
                <Text style={styles.formSection}>ADDRESS</Text>
                <View style={styles.formInputs}>
                    <InlineLabelTextInputField
                        name="country"
                        label="Country"
                    />
                    <InlineLabelTextInputField
                        name="state"
                        label="State"
                    />
                    <InlineLabelTextInputField
                        name="city"
                        label="City"
                    />
                    <InlineLabelTextInputField
                        name="address"
                        label="Address"
                    />
                    <InlineLabelTextInputField
                        name="zip"
                        label="ZIP code"
                    />
                </View>
                <Field name="save" component={Radio} />
                <Button
                    title="PAY"
                    onPress={this.onButtonPress}
                    containerViewStyle={styles.buttonContainer}
                    buttonStyle={styles.button}
                    textStyle={styles.buttonText}
                />
            </ScrollView>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff'
    },
    formSection: {
        fontSize: emY(0.8125),
        color: Color.GREY_600,
        paddingHorizontal: 15,
        paddingTop: emY(2.1875),
        paddingBottom: emY(1)
    },
    formInputs: {},
    buttonContainer: {},
    button: {
        backgroundColor: '#000',
        paddingVertical: emY(1)
    },
    buttonText: {
        fontSize: emY(0.8125)
    }
});

export default InformationSummary;
