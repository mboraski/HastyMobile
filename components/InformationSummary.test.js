import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import InformationSummary from './InformationSummary';

const initialState = {};

describe('InformationSummary', () => {
    const middlewares = [];
    const mockStore = configureStore(middlewares);
    it('renders correctly', () => {
        const wrapper = shallow(<InformationSummary />, {
            context: { store: mockStore(initialState) }
        });
        const render = wrapper.dive();
        expect(render).toMatchSnapshot();
    });
});
