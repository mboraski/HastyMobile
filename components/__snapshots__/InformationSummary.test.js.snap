// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationSummary renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Unknown
    style={
        Object {
            "backgroundColor": "#fff",
            "flex": 1,
          }
    }
>
    <View>
        <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                Object {
                    "color": "#7f7f7f",
                    "fontSize": 13,
                    "paddingBottom": 16,
                    "paddingHorizontal": 15,
                    "paddingTop": 35,
                  }
            }
        >
            INFORMATION
        </Text>
        <View
            style={Object {}}
        >
            <InlineLabelTextInputField
                label="Full Name"
                name="name"
            />
            <InlineLabelTextInputField
                label="Phone Number"
                name="number"
            />
            <InlineLabelTextInputField
                keyboardType="email-address"
                label="Email"
                name="email"
            />
        </View>
        <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                Object {
                    "color": "#7f7f7f",
                    "fontSize": 13,
                    "paddingBottom": 16,
                    "paddingHorizontal": 15,
                    "paddingTop": 35,
                  }
            }
        >
            ADDRESS
        </Text>
        <View
            style={Object {}}
        >
            <InlineLabelTextInputField
                label="Country"
                name="country"
            />
            <InlineLabelTextInputField
                label="State"
                name="state"
            />
            <InlineLabelTextInputField
                label="City"
                name="city"
            />
            <InlineLabelTextInputField
                label="Address"
                name="address"
            />
            <InlineLabelTextInputField
                label="ZIP code"
                name="zip"
            />
        </View>
        <Field
            component={[Function]}
            name="save"
        />
        <Button
            buttonStyle={
                Object {
                    "backgroundColor": "#000",
                    "paddingVertical": 16,
                  }
            }
            containerViewStyle={Object {}}
            onPress={undefined}
            textStyle={
                Object {
                    "fontSize": 13,
                  }
            }
            title="PAY"
        />
    </View>
</Unknown>,
  "nodes": Array [
    <Unknown
      style={
            Object {
                  "backgroundColor": "#fff",
                  "flex": 1,
                }
      }
>
      <View>
            <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                        Object {
                              "color": "#7f7f7f",
                              "fontSize": 13,
                              "paddingBottom": 16,
                              "paddingHorizontal": 15,
                              "paddingTop": 35,
                            }
                  }
            >
                  INFORMATION
            </Text>
            <View
                  style={Object {}}
            >
                  <InlineLabelTextInputField
                        label="Full Name"
                        name="name"
                  />
                  <InlineLabelTextInputField
                        label="Phone Number"
                        name="number"
                  />
                  <InlineLabelTextInputField
                        keyboardType="email-address"
                        label="Email"
                        name="email"
                  />
            </View>
            <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                        Object {
                              "color": "#7f7f7f",
                              "fontSize": 13,
                              "paddingBottom": 16,
                              "paddingHorizontal": 15,
                              "paddingTop": 35,
                            }
                  }
            >
                  ADDRESS
            </Text>
            <View
                  style={Object {}}
            >
                  <InlineLabelTextInputField
                        label="Country"
                        name="country"
                  />
                  <InlineLabelTextInputField
                        label="State"
                        name="state"
                  />
                  <InlineLabelTextInputField
                        label="City"
                        name="city"
                  />
                  <InlineLabelTextInputField
                        label="Address"
                        name="address"
                  />
                  <InlineLabelTextInputField
                        label="ZIP code"
                        name="zip"
                  />
            </View>
            <Field
                  component={[Function]}
                  name="save"
            />
            <Button
                  buttonStyle={
                        Object {
                              "backgroundColor": "#000",
                              "paddingVertical": 16,
                            }
                  }
                  containerViewStyle={Object {}}
                  onPress={undefined}
                  textStyle={
                        Object {
                              "fontSize": 13,
                            }
                  }
                  title="PAY"
            />
      </View>
</Unknown>,
  ],
  "options": Object {
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "_currentElement": <ScrollViewMock
        style={
                Object {
                        "backgroundColor": "#fff",
                        "flex": 1,
                      }
        }
>
        <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                        Object {
                                "color": "#7f7f7f",
                                "fontSize": 13,
                                "paddingBottom": 16,
                                "paddingHorizontal": 15,
                                "paddingTop": 35,
                              }
                }
        >
                INFORMATION
        </Text>
        <View
                style={Object {}}
        >
                <InlineLabelTextInputField
                        label="Full Name"
                        name="name"
                />
                <InlineLabelTextInputField
                        label="Phone Number"
                        name="number"
                />
                <InlineLabelTextInputField
                        keyboardType="email-address"
                        label="Email"
                        name="email"
                />
        </View>
        <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                        Object {
                                "color": "#7f7f7f",
                                "fontSize": 13,
                                "paddingBottom": 16,
                                "paddingHorizontal": 15,
                                "paddingTop": 35,
                              }
                }
        >
                ADDRESS
        </Text>
        <View
                style={Object {}}
        >
                <InlineLabelTextInputField
                        label="Country"
                        name="country"
                />
                <InlineLabelTextInputField
                        label="State"
                        name="state"
                />
                <InlineLabelTextInputField
                        label="City"
                        name="city"
                />
                <InlineLabelTextInputField
                        label="Address"
                        name="address"
                />
                <InlineLabelTextInputField
                        label="ZIP code"
                        name="zip"
                />
        </View>
        <Field
                component={[Function]}
                name="save"
        />
        <Button
                buttonStyle={
                        Object {
                                "backgroundColor": "#000",
                                "paddingVertical": 16,
                              }
                }
                containerViewStyle={Object {}}
                onPress={undefined}
                textStyle={
                        Object {
                                "fontSize": 13,
                              }
                }
                title="PAY"
        />
</ScrollViewMock>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ScrollViewMock {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                            Object {
                                          "color": "#7f7f7f",
                                          "fontSize": 13,
                                          "paddingBottom": 16,
                                          "paddingHorizontal": 15,
                                          "paddingTop": 35,
                                        }
              }
>
              INFORMATION
</Text>,
            <View
              style={Object {}}
>
              <InlineLabelTextInputField
                            label="Full Name"
                            name="name"
              />
              <InlineLabelTextInputField
                            label="Phone Number"
                            name="number"
              />
              <InlineLabelTextInputField
                            keyboardType="email-address"
                            label="Email"
                            name="email"
              />
</View>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                            Object {
                                          "color": "#7f7f7f",
                                          "fontSize": 13,
                                          "paddingBottom": 16,
                                          "paddingHorizontal": 15,
                                          "paddingTop": 35,
                                        }
              }
>
              ADDRESS
</Text>,
            <View
              style={Object {}}
>
              <InlineLabelTextInputField
                            label="Country"
                            name="country"
              />
              <InlineLabelTextInputField
                            label="State"
                            name="state"
              />
              <InlineLabelTextInputField
                            label="City"
                            name="city"
              />
              <InlineLabelTextInputField
                            label="Address"
                            name="address"
              />
              <InlineLabelTextInputField
                            label="ZIP code"
                            name="zip"
              />
</View>,
            <Field
              component={[Function]}
              name="save"
/>,
            <Button
              buttonStyle={
                            Object {
                                          "backgroundColor": "#000",
                                          "paddingVertical": 16,
                                        }
              }
              containerViewStyle={Object {}}
              onPress={undefined}
              textStyle={
                            Object {
                                          "fontSize": 13,
                                        }
              }
              title="PAY"
/>,
          ],
          "style": Object {
            "backgroundColor": "#fff",
            "flex": 1,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Unknown
          style={
                    Object {
                              "backgroundColor": "#fff",
                              "flex": 1,
                            }
          }
>
          <View>
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#7f7f7f",
                                                  "fontSize": 13,
                                                  "paddingBottom": 16,
                                                  "paddingHorizontal": 15,
                                                  "paddingTop": 35,
                                                }
                              }
                    >
                              INFORMATION
                    </Text>
                    <View
                              style={Object {}}
                    >
                              <InlineLabelTextInputField
                                        label="Full Name"
                                        name="name"
                              />
                              <InlineLabelTextInputField
                                        label="Phone Number"
                                        name="number"
                              />
                              <InlineLabelTextInputField
                                        keyboardType="email-address"
                                        label="Email"
                                        name="email"
                              />
                    </View>
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#7f7f7f",
                                                  "fontSize": 13,
                                                  "paddingBottom": 16,
                                                  "paddingHorizontal": 15,
                                                  "paddingTop": 35,
                                                }
                              }
                    >
                              ADDRESS
                    </Text>
                    <View
                              style={Object {}}
                    >
                              <InlineLabelTextInputField
                                        label="Country"
                                        name="country"
                              />
                              <InlineLabelTextInputField
                                        label="State"
                                        name="state"
                              />
                              <InlineLabelTextInputField
                                        label="City"
                                        name="city"
                              />
                              <InlineLabelTextInputField
                                        label="Address"
                                        name="address"
                              />
                              <InlineLabelTextInputField
                                        label="ZIP code"
                                        name="zip"
                              />
                    </View>
                    <Field
                              component={[Function]}
                              name="save"
                    />
                    <Button
                              buttonStyle={
                                        Object {
                                                  "backgroundColor": "#000",
                                                  "paddingVertical": 16,
                                                }
                              }
                              containerViewStyle={Object {}}
                              onPress={undefined}
                              textStyle={
                                        Object {
                                                  "fontSize": 13,
                                                }
                              }
                              title="PAY"
                    />
          </View>
</Unknown>,
        "_debugID": 4,
        "_renderedOutput": <Unknown
          style={
                    Object {
                              "backgroundColor": "#fff",
                              "flex": 1,
                            }
          }
>
          <View>
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#7f7f7f",
                                                  "fontSize": 13,
                                                  "paddingBottom": 16,
                                                  "paddingHorizontal": 15,
                                                  "paddingTop": 35,
                                                }
                              }
                    >
                              INFORMATION
                    </Text>
                    <View
                              style={Object {}}
                    >
                              <InlineLabelTextInputField
                                        label="Full Name"
                                        name="name"
                              />
                              <InlineLabelTextInputField
                                        label="Phone Number"
                                        name="number"
                              />
                              <InlineLabelTextInputField
                                        keyboardType="email-address"
                                        label="Email"
                                        name="email"
                              />
                    </View>
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#7f7f7f",
                                                  "fontSize": 13,
                                                  "paddingBottom": 16,
                                                  "paddingHorizontal": 15,
                                                  "paddingTop": 35,
                                                }
                              }
                    >
                              ADDRESS
                    </Text>
                    <View
                              style={Object {}}
                    >
                              <InlineLabelTextInputField
                                        label="Country"
                                        name="country"
                              />
                              <InlineLabelTextInputField
                                        label="State"
                                        name="state"
                              />
                              <InlineLabelTextInputField
                                        label="City"
                                        name="city"
                              />
                              <InlineLabelTextInputField
                                        label="Address"
                                        name="address"
                              />
                              <InlineLabelTextInputField
                                        label="ZIP code"
                                        name="zip"
                              />
                    </View>
                    <Field
                              component={[Function]}
                              name="save"
                    />
                    <Button
                              buttonStyle={
                                        Object {
                                                  "backgroundColor": "#000",
                                                  "paddingVertical": 16,
                                                }
                              }
                              containerViewStyle={Object {}}
                              onPress={undefined}
                              textStyle={
                                        Object {
                                                  "fontSize": 13,
                                                }
                              }
                              title="PAY"
                    />
          </View>
</Unknown>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ScrollViewMock
    style={
        Object {
            "backgroundColor": "#fff",
            "flex": 1,
          }
    }
>
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
            Object {
                "color": "#7f7f7f",
                "fontSize": 13,
                "paddingBottom": 16,
                "paddingHorizontal": 15,
                "paddingTop": 35,
              }
        }
    >
        INFORMATION
    </Text>
    <View
        style={Object {}}
    >
        <InlineLabelTextInputField
            label="Full Name"
            name="name"
        />
        <InlineLabelTextInputField
            label="Phone Number"
            name="number"
        />
        <InlineLabelTextInputField
            keyboardType="email-address"
            label="Email"
            name="email"
        />
    </View>
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
            Object {
                "color": "#7f7f7f",
                "fontSize": 13,
                "paddingBottom": 16,
                "paddingHorizontal": 15,
                "paddingTop": 35,
              }
        }
    >
        ADDRESS
    </Text>
    <View
        style={Object {}}
    >
        <InlineLabelTextInputField
            label="Country"
            name="country"
        />
        <InlineLabelTextInputField
            label="State"
            name="state"
        />
        <InlineLabelTextInputField
            label="City"
            name="city"
        />
        <InlineLabelTextInputField
            label="Address"
            name="address"
        />
        <InlineLabelTextInputField
            label="ZIP code"
            name="zip"
        />
    </View>
    <Field
        component={[Function]}
        name="save"
    />
    <Button
        buttonStyle={
            Object {
                "backgroundColor": "#000",
                "paddingVertical": 16,
              }
        }
        containerViewStyle={Object {}}
        onPress={undefined}
        textStyle={
            Object {
                "fontSize": 13,
              }
        }
        title="PAY"
    />
</ScrollViewMock>,
}
`;
